New merge command

lukasgierth opened the pull request:
Hi,

i added a simple merge command which simply calls "git merge $argv".
I use git merge quite often and was frustrated that i always have to use "git merge" instead of just "merge".

This first entry right now is only like an alias basically. But the plan is that the functionality can be expanded later on (dry runs etc.). I did not come up with a keyboard shortcut (that makes sense) yet.

Greetings from Germany,
stay healthy everyone!

lukasgierth committed:
add rudimantary merge command with simple branch completion

lukasgierth committed:
add manual entry (without keyboard shortcut)

joseluisq commented:
Hi, that's no bad idea to have the ease of typing during a merge via Fish auto completion.
However I would like to see some usage examples of that command (with no arguments and with them).

joseluisq commented on conf.d/gitnow.fish:
```diff
@@ -245,6 +245,22 @@ function release -d "GitNow: Creates a new Gitflow release branch from current b
+    end
+
+    set -l len (count $argv)
+    set -l opts .
```
Since the default argument looks like is a dot, a simple `merge` (`git merge .`) execution throws an error like `fatal: . - not something we can merge`.
If some argument is needed, I suggest to add some checks.

joseluisq reviewed (commented):
Also I would like to see some updates on the `README` for this new command with some examples.

lukasgierth committed:
do some checks, add abort option to merge, update README

lukasgierth commented:
I added some checks (does the branch exist, is it not the already active one etc.). I also added the -a option (instead of the branch  name) to abort a conflicted branch.
Right now it only checks if zero arguments are given. Should we better check that exactly one is give (so either a branch or the -a option)?
I also updated the README.

Some examples:
`❯ merge sss`
`Local branch sss was not found. Not possible to merge.`

`❯ merge`
`Merge: No argument given, needs one parameter`

`❯ merge master`
`Branch master is the same as current branch. Nothing to do,`

joseluisq commented:
> Right now it only checks if zero arguments are given. Should we better check that exactly one is give (so either a branch or the -a option)?

It's good idea since you need at least one arg.
However It will be even better if you make a distinguish between arguments like option flags (with dashes) and just branch names.
As a hint you can check the [switch statement](https://github.com/joseluisq/gitnow/blob/master/conf.d/gitnow.fish#L385) on `tag` command which is relative easy to implement.

lukasgierth commented:
If you look at my last commit, I already do use switch like you suggested

joseluisq commented:
Yes, you already do it. Sorry I had not seen it.

BTW Check this bugfix for https://github.com/joseluisq/gitnow/pull/28 which prevent an issue with the `__gitnow_check_if_branch_exist`.

joseluisq commented:
I plan to merge this in few minutes after a last review.

lukasgierth committed:
Merge pull request #1 from joseluisq/master

track changes from original repository

lukasgierth committed:
fix naming, merge not move

joseluisq commented:
When I try the `merge` command with option arguments (flags) then I get this:

```sh
merge --continue
# Local branch `` was not found. Not possible to merge.
```

I think you need to iterate over the `$argv` in order to know if you got a branch name param with options or without them.
The way to do it is implemented on move command.

https://github.com/joseluisq/gitnow/blob/708a95ae720441b115ac533babba373c62a832fb/conf.d/gitnow.fish#L256-L281

Which means you will need another check when a branch is not provided.
https://github.com/joseluisq/gitnow/blob/708a95ae720441b115ac533babba373c62a832fb/conf.d/gitnow.fish#L283-L289

lukasgierth committed:
add continue as option, capture unsupported --options

lukasgierth commented:
Pushed a new commit which should satisfy your suggestions.

joseluisq commented on conf.d/gitnow.fish:
```diff
@@ -245,6 +245,88 @@ function release -d "GitNow: Creates a new Gitflow release branch from current b
+
+    # No branch defined
+    if not test -n "$v_branch"
+        echo "Provide a valid branch name to switch to."
```
Just one more thing, please change `echo "Provide a valid branch name to switch to."` with `echo "Provide a valid branch name to merge."`

Then It should be ready to merge.

lukasgierth replied:
Done.

joseluisq commented:
I have tested and works fine! 

```sh
~> merge --continue
Continue the current merge
fatal: There is no merge in progress (MERGE_HEAD missing).
~> merge
Merge: No argument given, needs one parameter
~> merge -a
Abort the current merge
fatal: There is no merge to abort (MERGE_HEAD missing).
~> git merge --help
~> merge --continue --no-verify
Continue the current merge
fatal: There is no merge in progress (MERGE_HEAD missing).
~> merge --no-verify --continue
Continue the current merge
fatal: There is no merge in progress (MERGE_HEAD missing).
~> merge --no-verify
Provide a valid branch name to switch to
```

joseluisq reviewed (commented):
There is the last remaining change that should happen in order to merge this PR.

lukasgierth committed:
fix echo message

joseluisq reviewed (approved):
Thanks!
